src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala

```diff
@@ -1061,11 +1055,38 @@ object ModelFactory {
   val defaultGroupDesc = None
   val defaultGroupPriority = 1000
 
-  val tokens = raw"€\{($FILE_PATH|$FILE_EXT|$FILE_PATH_EXT|$FILE_LINE|$TPL_OWNER|$TPL_NAME)\}".r
+  // 4 subgroups: (.?), (FILE_PATH|FILE_PATH_EXT), (FILE_EXT) and (FILE_LINE|TPL_OWNER|TPL_NAME). non-matching are `null`
+  val tokens = raw"(?:(.?)€\{($FILE_PATH|$FILE_PATH_EXT)\}|\.?€\{($FILE_EXT)\}|€\{($FILE_LINE|$TPL_OWNER|$TPL_NAME)\})".r
```

Comment:
The outermost parens are extraneous.

The pattern and usage is clever but complex. You can look left given `m.start` to inspect the char, boring but clear.

Alternatively, or alternately, `\w(FILE_PATH)|\W(FILE_PATH)`. `(?<!Z)(?<bees>B+)` a group name to easily distinguish which match happened. `beez` would have the `Z`, pun alert.